Replace a load of references to the wiki with docs links

aydun opened the pull request:
Overview
----------------------------------------
Change some code comments & help links from the old wiki to new docs.
Change docs link from 'stable' to 'latest' (avoids redirection)

eileenmcnaughton commented:
@aydun looks like you need to negotiate a bit with jenkins

@MikeyMJCO can you review this?

MikeyMJCO commented:
Looks good to me!

colemanw commented:
@aydun the style warnings were:

```
Job.php:59, EndLine, Priority: High
--
Whitespace found at end of line


Job.php:73, EndLine, Priority: High
--
Whitespace found at end of line


Job.php:73, EmptyLines, Priority: High
--
Functions must not contain multiple empty lines in a row; found 2 empty lines
```

eileenmcnaughton commented:
I think with the @MikeyMJCO stamp of approval you just need the jenkins one & this is mergeable

eileenmcnaughton commented:
ok-without-test as only comment blocks involved

MikeyMJCO commented:
There are non-comment changes - to the help link destinations/code.

seamuslee001 commented on CRM/Utils/Check/Component/Env.php:
```diff
@@ -262,10 +262,9 @@ public function checkDefaultMailbox() {
       $message->addHelp(
         ts('A default mailbox must be configured for email bounce processing.') . '<br />' .
-        ts("Learn more in the <a %1>online documentation</a>.", [1 => $docUrl])
+          CRM_Utils_System::docURL2('user/advanced-configuration/email-system-configuration/')
```
Are we loosing the original link text here?

aydun replied:
Yes ... in some places we have things like this one: "Learn more in the online documentation" with a link, other places "See the XXX section in the documentation" with a link, others use the default rendering of "(Learn more ...)" with a link.  I've adopted the more concise, default form. 

So this one changes from:
"A default mailbox must be configured for email bounce processing. Learn more in the <a>online documentation</a>."
to
 "A default mailbox must be configured for email bounce processing. <a>(Learn more...)</a>"

MikeyMJCO replied:
I prefer the more concise form.

MikeyMJCO replied:
However I can't see where the "Learn more..." text is being passed in here.

MikeyMJCO replied:
Nope, I see it - in the `docUrl` functions.

aydun replied:
It's generated by docURL2()

seamuslee001 commented on CRM/Utils/Check/Component/Env.php:
```diff
@@ -606,7 +605,7 @@ public function checkExtensions() {
         __FUNCTION__,
         ts('There are currently no extensions on the CiviCRM public extension directory which are compatible with version %1. If you want to install an extension which is not marked as compatible, you may be able to <a %2>download and install extensions manually</a> (depending on access to your web server).', [
```
I think this will need to be changed so its 
```suggestion
        ts('There are currently no extensions on the CiviCRM public extension directory which are compatible with version %1. If you want to install an extension which is not marked as compatible, you may be able to <a href="%2">download and install extensions manually</a> (depending on access to your web server).', [
```

demeritcowboy replied:
I think the original is correct, or at least the civi standard, because it allows adding/removing/changing the attributes without breaking already translated strings. See https://docs.civicrm.org/dev/en/latest/translation/#avoid-tags-inside-strings where it talks about hyperlinks inside big text blocks.

seamuslee001 replied:
I guess my thinking is that wit the proposed change the href= is getting droped in param 2 so it needs to either get restored there or be added here

demeritcowboy replied:
Oh ok I see. Then it would be `<a %2>` and the href would be added into array param 2.

aydun replied:
You're right, it's lost the href.  Changed now to default form

seamuslee001 commented on templates/CRM/Admin/Form/Setting/Mail.tpl:
```diff
@@ -7,9 +7,8 @@
-{capture assign=docLink}{docURL page="CiviMail Mailer Settings" text="CiviMail Mailer Settings and Optimization" resource="wiki"}{/capture}
 <div class="help">
-    {ts 1=$docLink}These settings are used to configure mailer properties for the optional CiviMail component and may allow you to significantly optimize performance. Please read the %1 documentation, and make sure you understand it before modifying default values. (These settings are NOT used for the built-in 'Email - send now' feature).{/ts}
+    {ts}These settings are used to configure mailer properties for the optional CiviMail component and may allow you to significantly optimize performance. Please read the documentation, and make sure you understand it before modifying default values. (These settings are NOT used for the built-in 'Email - send now' feature).{/ts} {docURL page="sysadmin/setup/civimail/outbound"}
```
should we not still be passing through the text="xxx"here?

aydun replied:
See comment above

seamuslee001 commented on templates/CRM/Admin/Page/Options.hlp:
```diff
@@ -11,9 +11,8 @@
   {capture assign=crmURL}{crmURL p='civicrm/admin/custom/group' q='reset=1'}{/capture}
   <p>{ts 1=$crmURL}Subject, location, date/time and description fields are provided for all activity types. You can add custom fields for tracking additional information about activities <a href='%1'>here</a>.{/ts}</p>
-  <p>{ts 1=$docLink}Scheduled and Completed Activities are searchable by type and/or activity date using 'Advanced Search'. Other applications may record activities for CiviCRM contacts using our APIs. For more information, refer to the online %1.{/ts}</p>
+  <p>{ts}Scheduled and Completed Activities are searchable by type and/or activity date using 'Advanced Search'. Other applications may record activities for CiviCRM contacts using our APIs.{/ts} {docURL page="dev/api"} </p>
```
Again should we still be passing through the text here?

aydun replied:
See comment above

seamuslee001 commented:
I've added a few comments on where I think things could be improved

aydun commented:
Fixed the style issues (although not on lines I originally changed).  Changed CRM/Utils/Check/Component/Env.php and standardised a few more messages there.

colemanw commented:
Looks like all the review comments have been addressed and jenkins is happy.
